### PR TITLE
RWA-3300: Received messages help check repo tests added

### PR DESCRIPTION
**Pull request by adityadwadasi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-3300

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/resources/scripts/insert_case_event_messages_for_recived_messages_test.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+TRUNCATE TABLE public.wa_case_event_messages;
```
Comment (raghera) [inline src/integrationTest/resources/scripts/insert_case_event_messages_for_recived_messages_test.sql:1]: This file seems to have a typo?

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 2]
```diff
@@ -129,6 +129,42 @@ void test_health_reports_up_if_time_outside_of_working_hours_end_time() throws E
...
+
+        // THEN
+        mockMvc.perform(get("/health"))
+            .andExpect(
+                jsonPath("$.components.ccdMessagesReceived.status")
+                    .value(DOWN.toString()))
+            .andExpect(jsonPath("$.components.ccdMessagesReceived.details." + CASE_EVENT_HANDLER_MESSAGE_HEALTH)
+                           .value(String.format(NO_MESSAGES_RECEIVED, "test")));
+
+    }
```
Comment (raghera) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:166]: Did we also want to check the timezone issue in integration tests too or is this covered elsewhere?

FYI @BenNzewi

File: src/integrationTest/resources/scripts/insert_case_event_messages_for_recived_messages_test.sql (hunk 1 above)
Comment (adityadwadasi) [reply src/integrationTest/resources/scripts/insert_case_event_messages_for_recived_messages_test.sql]: THank you Ravi. I made necessary changes.

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java (hunk 2 above)
Comment (adityadwadasi) [reply src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:170]: I added 2 integration tests for timesone scenarios Ravi. Thank you.

**Review by raghera — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 3]
```diff
@@ -129,6 +133,85 @@ void test_health_reports_up_if_time_outside_of_working_hours_end_time() throws E
         verify(caseEventMessageRepository, never()).getNumberOfMessagesReceivedInLastHour(any());
     }
 
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:scripts/insert_case_event_messages_for_received_messages_check.sql"})
+    void test_health_reports_up_if_received_messages_in_last_hour() throws Exception {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:139]: can we update this to use Should_ExpectedBehavior_When_StateUnderTest naming convention?

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 4]
```diff
@@ -129,6 +133,85 @@ void test_health_reports_up_if_time_outside_of_working_hours_end_time() throws E
...
+                    .value(UP.toString()))
+            .andExpect(jsonPath("$.components.ccdMessagesReceived.details." + CASE_EVENT_HANDLER_MESSAGE_HEALTH)
+                           .value(String.format(MESSAGES_RECEIVED, "test")));
+
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:scripts/insert_case_event_messages_for_received_messages_check.sql"})
+    void test_health_reports_down_if_not_received_messages_in_last_hour() throws Exception {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:157]: can we update this to use Should_ExpectedBehavior_When_StateUnderTest naming convention?

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 5]
```diff
@@ -129,6 +133,85 @@ void test_health_reports_up_if_time_outside_of_working_hours_end_time() throws E
...
+                jsonPath("$.components.ccdMessagesReceived.status")
+                    .value(DOWN.toString()))
+            .andExpect(jsonPath("$.components.ccdMessagesReceived.details." + CASE_EVENT_HANDLER_MESSAGE_HEALTH)
+                           .value(String.format(NO_MESSAGES_RECEIVED, "test")));
+
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "workingHoursWithTimeZoneScenarioProvider")
+    void test_health_calls_repository_if_working_day_time_is_within_working_hours_with_timezone(
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:174]: can we update this to use Should_ExpectedBehavior_When_StateUnderTest naming convention?

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 6]
```diff
@@ -129,6 +133,85 @@ void test_health_reports_up_if_time_outside_of_working_hours_end_time() throws E
...
+        when(caseEventMessageRepository.getNumberOfMessagesReceivedInLastHour(any())).thenReturn(10);
+
+        // THEN
+        assertReceivedMessagesHealthStatus(UP, MESSAGES_RECEIVED);
+
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "nonWorkingHoursForDstTimeZoneStartTimeAndEndTime")
+    void test_health_calls_repository_if_working_day_time_is_outside_working_hours_with_timezone(
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:189]: can we update this to use Should_ExpectedBehavior_When_StateUnderTest naming convention?

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 7]
```diff
@@ -97,7 +101,7 @@ void test_health_reports_up_if_no_messages_received_in_last_hour_during_weekend(
     @Test
     void test_health_reports_up_if_no_messages_received_in_last_hour_during_holiday() throws Exception {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:102]: this needs renaming as well

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java [hunk 8]
```diff
@@ -129,6 +133,85 @@ void test_health_reports_up_if_time_outside_of_working_hours_end_time() throws E
...
+                    .value(UP.toString()))
+            .andExpect(jsonPath("$.components.ccdMessagesReceived.details." + CASE_EVENT_HANDLER_MESSAGE_HEALTH)
+                           .value(String.format(MESSAGES_RECEIVED, "test")));
+
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:scripts/insert_case_event_messages_for_received_messages_check.sql"})
+    void should_health_check_reports_down_when_no_messages_received_in_last_hour() throws Exception {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java:157]: should_verify_health_check_reports_is_down_when_no_messages_received_in_last_hour?

**Review by BenNzewi — APPROVED** [review]

**Pull request merged**